website/src/components/pageStateSelectors/DynamicDateFilter.tsx

```diff
@@ -0,0 +1,111 @@
+import { type DateRangeOption } from '@genspectrum/dashboard-components/util';
+import { useQuery } from '@tanstack/react-query';
+import dayjs from 'dayjs';
+import React from 'react';
+
+import { Loading } from '../../util/Loading';
+import { GsDateRangeFilter } from '../genspectrum/GsDateRangeFilter';
+import { withQueryProvider } from '../subscriptions/backendApi/withQueryProvider';
+
+function DynamicDateFilterInner({
+    label,
+    lapis,
+    dateFieldName,
+    baselineOptions,
+    value,
+    onChange,
+}: {
+    label: string;
+    lapis: string;
+    dateFieldName: string;
+    baselineOptions: DateRangeOption[];
+    value: DateRangeOption | undefined;
+    onChange: (newValue: DateRangeOption | undefined) => void;
+}) {
+    const {
+        data: dateRange,
+        isPending,
+        isError,
+        error,
+    } = useQuery({
+        queryKey: ['dateRange', lapis, dateFieldName],
+        queryFn: () => fetchDateRange(lapis, dateFieldName),
+    });
+
+    return (
+        <label className='form-control'>
+            <div className='label'>
+                <span className='label-text'>{label}</span>
+            </div>
+            {isPending ? (
+                <div className='h-20'>
+                    <Loading />
+                </div>
+            ) : isError ? (
+                <div className='flex h-20 items-center'>
+                    Failed to load date range: {error instanceof Error ? error.message : String(error)}
+                </div>
+            ) : (
+                <GsDateRangeFilter
+                    lapisDateField={dateFieldName}
+                    onDateRangeChange={(dateRange: DateRangeOption | null) => onChange(dateRange ?? undefined)}
+                    value={value}
+                    dateRangeOptions={filteredOptions(baselineOptions, dateRange.start, dateRange.end)}
+                />
+            )}
+        </label>
+    );
+}
+
+/**
+ * The `DynamicDateFilter` computes the available date range options dynamically,
+ * based on the available dates in the data, for the given LAPIS and field name.
+ */
+export const DynamicDateFilter = withQueryProvider(DynamicDateFilterInner);
```

Comment:
The `WasapPage` already has a query provider. I think we don't need it here, do we?